### ISSUE TITLE
Distinct fail when changing index

### ISSUE THREAD

**Issue by fvoverscan** [algolia/instantsearch.js#3270]
Hi,

I have an issue when i use a sortBySelector widget...
When i change the indice, the distinct parameter don't work anymore and i have more hits then i expect to.

I have this configuration :
`const search = instantsearch({
            appId: client.applicationID,
            apiKey: client.apiKey,
            indexName: index.indexName,
            searchParameters: {
                hitsPerPage: 15,
                distinct: 1,
                facetingAfterDistinct: true,
            },
            routing: true
        });`

`search.addWidget(
            instantsearch.widgets.sortBySelector({
                container: '#sort-product',
                indices: [
                    {name: index.indexName, label: 'Last modified'},
                    {name: index.indexName+'price_with_vat', label: 'Lower to Higher Price)'},
                    {name: index.indexName+'-price_with_vat', label: 'Higher to Lower Price)'}
                ]
            })
        );`

You could test the issue on that site (french only) :
https://www.maxxess.fr/categorie/huiles-2t-226

Is there something in the configuration that i miss ?

Thanks

**Comment by fvoverscan** [comment]
Is it possible that the settings i set to the index are not applied to the slave index ?

**Comment by samouss** [comment]
I didn't manage to reproduce the issue on your website from the provided URL. I switch between the three indices but I always get the same number of results. Did I miss something?

**Comment by fvoverscan** [comment]
Yes i apply the settings on all slaves index too so the problem is resolved.
But i would like to know if this behavior is normal ?

**Comment by samouss** [comment]
If I understand correctly the issue was: the settings were not applied on the replicas right? Yes it's the expected behaviour. You have to provide an options to apply the [`setSettings`](https://www.algolia.com/doc/api-reference/api-methods/set-settings/?language=javascript) call to the replicas. This option is [`forwardToReplicas `](https://www.algolia.com/doc/api-reference/api-methods/set-settings/?language=javascript#method-param-forwardtoreplicas).
